service/controller/clusterapi/v21/resource/certconfig/desired.go

```diff
@@ -27,6 +27,20 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 		return nil, microerror.Mask(err)
 	}
 
+	// When the CertConfig CR is deleted we do not need to compute the desired
+	// state, because we only use the current state to delete the CR. Also note
+	// that teh desired state relies on the operatorversions resource, because we
+	// put the cert-operator version into the CR. The operatorversions resource
+	// does not fill the controller context with versions on delete events, which
+	// is also why we cannot compute the correct desired state. We do not want to
+	// fetch the version information on delete events to reduce eventual friction.
+	// Cluster deletion should not be affected only because some releases are
+	// missing or broken when fetching them from cluster-service.
+	if key.IsDeleted(&cr) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "not computing desired state of cert config crs due to delete event")
+		return nil, nil
+	}
```

Comment:
The version information are only used by the `app` and `certconfig` resource. The former already cancels on delete. We want to delete CertConfig CRs though, so we just skip the desired state computation on delete events here.